crates/rpc/rpc-eth-types/src/cache/mod.rs

```diff
@@ -61,15 +61,10 @@ type HeaderLruCache<H, L> = MultiConsumerLruCache<B256, H, L, HeaderResponseSend
 ///
 /// This is the frontend for the async caching service which manages cached data on a different
 /// task.
-#[derive(Debug)]
+#[derive(Debug, Clone)]
 pub struct EthStateCache<B: Block, R> {
     to_service: UnboundedSender<CacheAction<B, R>>,
-}
-
-impl<B: Block, R> Clone for EthStateCache<B, R> {
-    fn clone(&self) -> Self {
-        Self { to_service: self.to_service.clone() }
-    }
+    latest_chain_change: Option<ChainChange<B, R>>,
```

Comment:
we don't need to share this and can instead rely on message passing and always fetch this from the service